### PR TITLE
8.0 backport barcodes

### PR DESCRIPTION
**Pull request by legalsylvain**
This PR backport 'barcodes' module, from odoo core in 9.0, to 8.0 serie.

if you want to use it, please review only the two last commits. (the first one is just a copy past of the 9.0 serie).

Main changes

- disable JS features
- add data from stock and point of sale (9.0)
- add noqa in the whole module, to let the possibility to meld the files, with the original code
- add odoo licence and copyright

Note : some pylint errors break travis. Feel free to make a fix against this PR, if you're interested.

### CODE REVIEW

**Comment by legalsylvain** [comment]
@gurneyalex : do you know how to disable pilint check ?

**Comment by gurneyalex** [comment]
@legalsylvain not sure how to do this (or if we want to...)

**Comment by legalsylvain** [comment]
well, the problem is that PR is a huge backport that come from Odoo 9.0 CE, mainly to use barcode generation in V8.0. https://github.com/OCA/stock-logistics-barcode/pull/92

I don't think that it's usefull to fix all the pylint error, written by the Odoo SA team. And it will make harder a diff review. Don't you think ?

**Comment by rafaelbn** [comment]
Hi @legalsylvain I don't think I can help here :cry:

File: .travis.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ env:
   - secure: jX9I8iBSTGMGYNzj5zJkh/bvmXmemjqlKpVezZ2QMqaoXHgPPUlktxBwTMlmNeuJ5L9DlDiCqOiXpF41LDmKmjX/e99G15cvNSfQc/jG+exIfo9KSbc6aQSu8cqP+yu1MGTsuuq67Wm8N45+fFJGGsySUbGWFPXnF4MVIiZpqx0=
 
   matrix:
-  - LINT_CHECK="1"
+  - LINT_CHECK="1" EXCLUDE="barcodes"
```
Comment (gurneyalex) [inline .travis.yml:31]: This won't work : the  EXCLUDE environment variable is only taken into account for tests, not for linting.

**Comment by legalsylvain** [comment]
seems I fucked my branch with a bad rebase. Well, I don't reach to disable pylint / flake8 check for this module.
closing this PR. 
If you want to use this module for 8.0 version, you can take it from this repository. 

https://github.com/grap/grap-odoo-backport/tree/8.0

regards.

**Pull request closed without merging**